pkg/controller/job/utils_test.go

```diff
@@ -46,4 +46,21 @@ func TestIsJobFinished(t *testing.T) {
 	if IsJobFinished(job) {
 		t.Error("Job was not expected to be finished")
 	}
+
+	job.Status.Conditions[0].Type = batch.JobFailed
+
+	job.Status.Conditions[0].Status = v1.ConditionTrue
+	if !IsJobFinished(job) {
+		t.Error("Job was expected to be finished")
+	}
+
+	job.Status.Conditions[0].Status = v1.ConditionFalse
+	if IsJobFinished(job) {
+		t.Error("Job was not expected to be finished")
+	}
+
+	job.Status.Conditions[0].Status = v1.ConditionUnknown
+	if IsJobFinished(job) {
+		t.Error("Job was not expected to be finished")
+	}
```

Comment:
Thanks a lot for the patch. Would you make these tests [table-driven](https://github.com/golang/go/wiki/TableDrivenTests) while you're at it?